File: pippip-native-ios/Support/AccountDeleter.m
//
//  AccountDeleter.m
//  pippip-native-ios
//
//  Created by Steve Brenneis on 4/3/18.
//  Copyright © 2018 seComm. All rights reserved.
//

#import "pippip_native_ios-Swift.h"
#import "AccountDeleter.h"
#import "ApplicationSingleton.h"
#import <Realm/Realm.h>

@implementation AccountDeleter

- (BOOL)deleteAccount:(NSString *)accountName {
    
    NSFileManager *manager = [NSFileManager defaultManager];
    RLMRealmConfiguration *config = [RLMRealmConfiguration defaultConfiguration];
    NSArray<NSURL *> *realmFileURLs = @[
                                        config.fileURL,
                                        [config.fileURL URLByAppendingPathExtension:@"lock"],
                                        [config.fileURL URLByAppendingPathExtension:@"note"],
                                        [config.fileURL URLByAppendingPathExtension:@"management"]
                                        ];
    for (NSURL *URL in realmFileURLs) {
        NSError *error = nil;
        [manager removeItemAtURL:URL error:&error];
        if (error) {
            // handle error
        }
    }
    
    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
    NSString *docPath = [paths objectAtIndex:0];
    NSString *vaultsPath = [docPath stringByAppendingPathComponent:@"PippipVaults"];
    NSString *vaultPath = [vaultsPath stringByAppendingPathComponent:accountName];
    return [manager removeItemAtPath:vaultPath error:nil];
    
}

@end

File: pippip-native-ios/Support/AccountDeleter.swift
//
//  AccountDeleter.swift
//  pippip-native-ios
//
//  Created by Steve Brenneis on 6/12/18.
//  Copyright © 2018 seComm. All rights reserved.
//

import RealmSwift
import CocoaLumberjack

class AccountDeleter: NSObject {

    func deleteAccount() {
        
        deleteDirectoryId()
        let accountName = AccountSession.instance.accountName
        
        do {
            let realmURL = Realm.Configuration.defaultConfiguration.fileURL!
            let realmURLs = [
                realmURL,
                realmURL.appendingPathExtension("lock"),
                realmURL.appendingPathExtension("note"),
                realmURL.appendingPathExtension("management")
            ]
            for URL in realmURLs {
                try FileManager.default.removeItem(at: URL)
            }
        }
        catch {
            DDLogError("Error while deleting Realm files: \(error.localizedDescription)")
        }
        
        do {
            let docsURLs = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask)
            let docURL = docsURLs[0]
            let vaultsURL = docURL.appendingPathComponent("PippipVaults", isDirectory: true)
            let vaultUrl = vaultsURL.appendingPathComponent(accountName)
            try FileManager.default.removeItem(at: vaultUrl)
        }
        catch {
            DDLogError("Error while deleting user vault: \(error.localizedDescription)")
        }
        
    }

    func deleteDirectoryId() {

        let config = Configurator()
        if let directoryId = config.directoryId {
            ContactManager.instance.updateDirectoryId(newDirectoryId: nil, oldDirectoryId: directoryId)
        }

    }

}
